packages/rust-dprint/src/main.rs

```diff
@@ -0,0 +1,213 @@
+extern crate dprint_core;
+extern crate dprint_plugin_typescript as dprint_ts;
+
+use clap::{App, Arg};
+use dprint_ts::configuration::{Configuration, ConfigurationBuilder};
+use serde_json;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn main() {
+    let cli_parser = create_cli_parser();
+
+    let matches = cli_parser.get_matches();
+
+    let config = if let Some(config_path) = matches.value_of("config") {
+        let config_contents = match fs::read_to_string(&config_path) {
+            Ok(contents) => contents,
+            Err(e) => {
+                eprintln!("{}", e.to_string());
+                std::process::exit(1);
+            }
+        };
+
+        let unresolved_config: HashMap<String, String> =
+            match serde_json::from_str(&config_contents) {
+                Ok(c) => c,
+                Err(e) => {
+                    eprintln!("{}", e.to_string());
+                    std::process::exit(1);
+                }
+            };
+
+        let global_config_result =
+            dprint_core::configuration::resolve_global_config(&unresolved_config.clone());
```

Comment:
Right now we don't have a global config, but in the future we could structure it like:

```jsonc
{
  // global config goes here (ex. "newLineKind": "auto")
  "typescript": {
    // typescript config goes here
  }
}
```

I think this should probably just pass in an empty hashmap in here for now.

Reply:
@bartlomieju I'm going to merge this change in later today after work. Other than this, it looks great! 😀

By the way, I only have a few issues remaining before I recommend using this on Deno's code: https://github.com/dsherret/dprint/milestone/2 -- Once those are done then it will be good to start using.

Reply:
@dsherret I'm not sure I understand... 
```json
{
    "lineWidth": "40"
}
```
^-- this config file works
```json
{
    "lineWidth": "40",
    "typescript": {}
}
```
^--- this config file doesn't, because `typescript` is a map not a string

I thought that different options can be mixed in a single file and running it through both global and TS config will pick up all options.

Reply:
> @bartlomieju I'm going to merge this change in later today after work. Other than this, it looks great! 😀
> 
> By the way, I only have a few issues remaining before I recommend using this on Deno's code: https://github.com/dsherret/dprint/milestone/2 -- Once those are done then it will be good to start using.

That's great! Waiting eagerly to replace prettier :))

Reply:
So there's "global config" that is config shared across all languages, then there's language specific config that's implemented in dprint-plugin-typescript.

For now we'll just have the typescript config at the top level like is here so it will be (as you've done here):

```ts
{
    "lineWidth": "40",
    "trailingCommas": "always"
}
```

In the future once we have a "global config" then we'll probably move the typescript config into a "typescript" property.

```ts
{
    "lineWidth": "40",
    "typescript": {
        "trailingCommas": "always"
    }
}
```

So since we don't have a global config at the moment in the json file, we should probably change this one line to `dprint_core::configuration::resolve_global_config(&new HashMap())`. Does that make sense?

Reply:
> I thought that different options can be mixed in a single file and running it through both global and TS config will pick up all options.

I think right now the `resolve_config` in `dprint-plugin-typescript` will pick up global config like `lineWidth` as well. The `resolve_config` only uses the global config for its defaults.

But yeah, now that I think about it, the code will work as-is right now so this is not a big deal.

Reply:
Okay, all clear, let me change that and link to this comment.

Reply:
Perfect! Thanks!

Yeah, here's a better example of what would work in the future:

```json
{
    "indentWidth": 4,
    "typescript": {
        "indentWidth": 2,
        "trailingCommas": "always"
    }
}
```